pkg/costmodel/allocation.go

```diff
@@ -1026,41 +1230,72 @@ func applyGPUsAllocated(podMap map[podKey]*Pod, resGPUsRequested []*prom.QueryRe
 			continue
 		}
 
-		pod, ok := podMap[key]
-		if !ok {
-			continue
-		}
-
 		container, err := res.GetString("container")
 		if err != nil {
 			log.DedupedWarningf(10, "CostModel.ComputeAllocation: GPU request query result missing 'container': %s", key)
 			continue
 		}
 
-		if _, ok := pod.Allocations[container]; !ok {
-			pod.AppendContainer(container)
+		var pods []*Pod
+
+		pod, ok := podMap[key]
+		if !ok {
+			if uidKeys, ok := podUIDKeyMap[key]; ok {
+				for _, uidKey := range uidKeys {
+					pod, ok = podMap[uidKey]
+					if ok {
+						pods = append(pods, pod)
+					}
+				}
+			} else {
+				continue
+			}
+		} else {
+			pods = []*Pod{pod}
 		}
 
-		hrs := pod.Allocations[container].Minutes() / 60.0
-		pod.Allocations[container].GPUHours = res.Values[0].Value * hrs
+		for _, pod := range pods {
+
+			if _, ok := pod.Allocations[container]; !ok {
+				pod.AppendContainer(container)
+			}
+
+			hrs := pod.Allocations[container].Minutes() / 60.0
+			pod.Allocations[container].GPUHours = res.Values[0].Value * hrs
+		}
 	}
 }
 
-func applyNetworkTotals(podMap map[podKey]*Pod, resNetworkTransferBytes []*prom.QueryResult, resNetworkReceiveBytes []*prom.QueryResult) {
+func applyNetworkTotals(podMap map[podKey]*Pod, resNetworkTransferBytes []*prom.QueryResult, resNetworkReceiveBytes []*prom.QueryResult, podUIDKeyMap map[podKey][]podKey) {
 	for _, res := range resNetworkTransferBytes {
 		podKey, err := resultPodKey(res, env.GetPromClusterLabel(), "namespace")
 		if err != nil {
 			log.DedupedWarningf(10, "CostModel.ComputeAllocation: Network Transfer Bytes query result missing field: %s", err)
 			continue
 		}
 
+		var pods []*Pod
+
 		pod, ok := podMap[podKey]
 		if !ok {
-			continue
+			if uidKeys, ok := podUIDKeyMap[podKey]; ok {
+				for _, uidKey := range uidKeys {
+					pod, ok = podMap[uidKey]
+					if ok {
+						pods = append(pods, pod)
+					}
+				}
+			} else {
+				continue
+			}
+		} else {
+			pods = []*Pod{pod}
 		}
 
-		for _, alloc := range pod.Allocations {
-			alloc.NetworkTransferBytes = res.Values[0].Value / float64(len(pod.Allocations))
+		for _, pod := range pods {
+			for _, alloc := range pod.Allocations {
+				alloc.NetworkTransferBytes = res.Values[0].Value / float64(len(pod.Allocations))
```

Comment:
Should this also be divided by `float64(len(pods))`?

Reply:
That would be "more correct" I think, as the query value is a summation. Good catch. Added.